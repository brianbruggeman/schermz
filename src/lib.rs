mod schema;
mod schema_object;
mod schema_object_key;
mod schema_value_type;
mod value_type;

pub use schema::Schema;
pub use schema_object::SchemaObject;
pub use schema_object_key::SchemaObjectKey;
pub use schema_value_type::SchemaValueType;
pub use value_type::ValueType;
