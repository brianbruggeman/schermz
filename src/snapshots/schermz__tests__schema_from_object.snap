---
source: src/lib.rs
assertion_line: 100
expression: "Schema::from_json(&json, true).to_json()"
---
{
  "address": {
    "types": [
      {
        "city": {
          "types": [
            "STRING(6)"
          ]
        },
        "street": {
          "types": [
            "STRING(17)"
          ]
        }
      }
    ]
  },
  "age": {
    "types": [
      "NUMBER"
    ]
  },
  "name": {
    "types": [
      "STRING(8)"
    ]
  },
  "phones": {
    "types": [
      {
        "ARRAY": [
          {
            "mobile": {
              "types": [
                "STRING(11)"
              ]
            }
          },
          "NUMBER",
          "STRING(11)"
        ]
      }
    ]
  },
  "title": {
    "types": [
      "STRING(0)"
    ]
  }
}
